plugins/Monolog/config/config.php

```
@@ -226,22 +227,70 @@
     }),
 
     'Piwik\Plugins\Monolog\Formatter\LineMessageFormatter' => Piwik\DI::create('Piwik\Plugins\Monolog\Formatter\LineMessageFormatter')
-                                                                ->constructor(Piwik\DI::get('log.short.format')),
+        ->constructor(Piwik\DI::get('log.short.format')),
     'log.lineMessageFormatter' => Piwik\DI::create('Piwik\Plugins\Monolog\Formatter\LineMessageFormatter')
         ->constructor(Piwik\DI::get('log.short.format')),
 
     'log.lineMessageFormatter.file' => Piwik\DI::autowire('Piwik\Plugins\Monolog\Formatter\LineMessageFormatter')
         ->constructor(Piwik\DI::get('log.trace.format'))
-        ->constructorParameter('allowInlineLineBreaks', false),
+        ->constructorParameter('allowInlineLineBreaks', false)
+        ->constructorParameter('customFunctionFile', Piwik\DI::get('log.custom_function_file'))
+        ->constructorParameter('excludePatterns', Piwik\DI::get('log.exclude_patterns')),
+
+    'log.exclude_patterns' => Piwik\DI::factory(function (Container $c) {
+        if ($c->has('ini.log.exclude_patterns')) {
+            $xcl = [];
+            foreach (explode("|", $c->get('ini.log.exclude_patterns')) as $p) {
+                $xcl[]  = trim($p);
+            }
+            return $xcl;
+        }
+        return null;
+    }),
+
+    'log.custom_function_file' => Piwik\DI::factory(function (Container $c) {
+        if ($c->has('ini.log.custom_function_file')) {
+            $path = $c->get('ini.log.custom_function_file');
+            if (!file_exists($path)) {
+                return null;
+                throw new \Exception('Specified path to custom_function_file does not exist: ' . $path);
```

Wont' be thrown if we return above it. Similar for the readability check.

---

@michalkleiner oh yeah that was test code I forgot to go back over.  What do you think / what is our general philosophy in this case?  Should the code fail silently or throw an exception and break logging?